### PR TITLE
[HOPS-1628] Use TLS for Yarn web UI

### PR DESCRIPTION
**Pull request by SirOibaf**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added and passed (for bug fixes / features)
- [x] HOPS JIRA issue has been opened for this PR
- [ ] All commits have been squashed down to a single commit
- [x] The commit message has the following format: [HOPS-XXX] message

* **Post a link to the associated JIRA issue**

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java [hunk 1]
```diff
@@ -428,18 +442,44 @@ public static String getHttpSchemePrefix(Configuration conf) {
...
+      return loadFromSSLServerConf(builder, sslConf);
+    }
+    return loadFromSuperUserLoader(builder, sslConf);
+  }
+
+  private static HttpServer2.Builder loadFromSSLServerConf(HttpServer2.Builder builder, Configuration sslConf) {
     return builder
-        .needsClientAuth(needsClientAuth)
-        .keyPassword(getPassword(sslConf, WEB_APP_KEY_PASSWORD_KEY))
+        .keyPassword(getPassword(sslConf, DFS_SERVER_HTTPS_KEYPASSWORD_KEY))
```
Comment (kouzant) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java:459]: Why replacing the property name with DFS variables? Same for the rest two below

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java [hunk 2]
```diff
@@ -428,18 +442,44 @@ public static String getHttpSchemePrefix(Configuration conf) {
...
+            getPassword(sslConf, DFS_SERVER_HTTPS_KEYSTORE_PASSWORD_KEY),
             sslConf.get("ssl.server.keystore.type", "jks"))
         .trustStore(sslConf.get("ssl.server.truststore.location"),
-            getPassword(sslConf, WEB_APP_TRUSTSTORE_PASSWORD_KEY),
-            sslConf.get("ssl.server.truststore.type", "jks"))
-        .excludeCiphers(
-            sslConf.get("ssl.server.exclude.cipher.list"));
+            getPassword(sslConf, DFS_SERVER_HTTPS_TRUSTSTORE_PASSWORD_KEY),
+            sslConf.get("ssl.server.truststore.type", "jks"));
+  }
```
Comment (kouzant) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java:466]: If it doesn't break anything, I would keep the `.excludeCiphers()`

**Review by kouzant — CHANGES_REQUESTED** [review]

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java (hunk 2 above)
Comment (SirOibaf) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java:466]: Done added also for the `loadFromSuperUserLoader`

**Comment by berthoug** [comment]
Jenkins test this please.

**Comment by berthoug** [comment]
Jenkins test this please.

**Comment by smkniazi** [comment]
Jenkins test this please.

**Comment by smkniazi** [comment]
Jenkins test this please.

**Comment by smkniazi** [comment]
Jenkins test this please.

**Comment by smkniazi** [comment]
Jenkins test this please.

**Comment by smkniazi** [comment]
Jenkins test this please.

**Comment by smkniazi** [comment]
Jenkins test this please.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java [hunk 3]
```diff
@@ -17,6 +17,11 @@
 */
 package org.apache.hadoop.yarn.webapp.util;
 
+import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_CLIENT_HTTPS_NEED_AUTH_DEFAULT;
+import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_CLIENT_HTTPS_NEED_AUTH_KEY;
+import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_SERVER_HTTPS_KEYPASSWORD_KEY;
+import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_SERVER_HTTPS_KEYSTORE_PASSWORD_KEY;
+import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_SERVER_HTTPS_TRUSTSTORE_PASSWORD_KEY;
```
Comment (kouzant) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java:24]: @SirOibaf I believe these imports are not relevant any more

**Review by kouzant — CHANGES_REQUESTED** [review]

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java (hunk 3 above)
Comment (SirOibaf) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java]: Removed

**Review by kouzant — APPROVED** [review]

**Comment by berthoug** [comment]
Jenkins test this please.

**Review by kouzant — APPROVED** [review]

**Pull request merged**